index.js

```diff
@@ -1,23 +1,24 @@
 /* jshint node: true */
 'use strict';
 
-var GithubAdapter = require('./lib/github');
-
-var nothing = function() {
-  return {
-    upload: function() {}
-  }
-}
+var Uploader = require('./lib/uploader');
+var DeployPluginBase = require('ember-cli-deploy-plugin');
 
 module.exports = {
-  name: 'ember-deploy-github',
-  type: 'ember-deploy-addon',
-  adapters: {
-    assets: {
-      'github': GithubAdapter
-    },
-    index: {
-      'github': nothing
-    }
+  name: 'ember-cli-deploy-github',
+  createDeployPlugin: function(options) {
+    var DeployPlugin = DeployPluginBase.extend({
+      name: options.name,
+      defaultConfig: {
+        branch: 'gh-pages'
+      },
+      requiredConfig: ['repository'],
+      upload: function (context) {
+        this.git = context.config.git || require('nodegit');
```

Comment:
Thank you! I think now is better, I move configuration logic to configure hooks.